### PR TITLE
[REF-6205, 6206] Improve SageMaker job metrics reporting

### PR DESCRIPTION
**Pull request by zohar-mizrahi**

### CODE REVIEW

File: mlcomp/parallelm/extra/sagemaker/monitor/job_monitor_base.py [hunk 1]
```diff
@@ -51,10 +57,40 @@ def set_on_complete_callback(self, on_complete_callback):
...
+    def _total_running_time_sec(self, describe_response):
+        create_time = self._job_create_time(describe_response)
+        if create_time is None:
+            return None
+
+        end_time = self._job_end_time(describe_response)
+        if end_time:
+            return (end_time - create_time).total_seconds()
+        else:
+            return (datetime.now(pytz.UTC) - create_time).total_seconds()
```
Comment (liororama) [inline mlcomp/parallelm/extra/sagemaker/monitor/job_monitor_base.py:69]: Is UTC the correct time Zone?

File: mlcomp/parallelm/extra/sagemaker/monitor/report.py [hunk 2]
```diff
@@ -8,10 +8,21 @@ class Report(object):
     _last_metric_values = {}
 
     @staticmethod
-    def job_status(job_name, running_time_sec, status):
+    def job_status(job_name, running_time_sec, billing_time_sec, status=""):
         Report._last_metric_values[job_name] = status
-        tbl = Table().name("SageMaker Job Status").cols(["Job Name", "Running Time (sec)", "Status"])
-        tbl.add_row([job_name, str(timedelta(seconds=running_time_sec)), status])
+        tbl = Table().name("SageMaker Job Status").cols(["Job Name", "Total Running Time",
+                                                         "Time for Billing", "Status"])
```
Comment (liororama) [inline mlcomp/parallelm/extra/sagemaker/monitor/report.py:14]: Regarding billing time, is it possible also to obtain the cost? (in $$)

**Review by liororama — APPROVED** [review]

File: mlcomp/parallelm/extra/sagemaker/monitor/job_monitor_base.py [hunk 3]
```diff
@@ -19,29 +21,33 @@ def __init__(self, sagemaker_client, job_name, logger):
...
             response = self._describe_job()
-            running_time_sec = int(time.time() - start_running_time_sec)
             if self._logger.isEnabledFor(logging.DEBUG):
                 self._logger.debug(pprint.pformat(response, indent=4))
 
             status = self._job_status(response)
-            Report.job_status(self._job_name, running_time_sec, status)
+            running_time_sec = self._total_running_time_sec(response)
+            billing_time_sec = self._billing_time_sec(response)
+            Report.job_status(self._job_name, running_time_sec, billing_time_sec, status)
```
Comment (yakov-g) [inline mlcomp/parallelm/extra/sagemaker/monitor/job_monitor_base.py:32]: Looks like `running_time_sec`, `billing_time_sec` not likely, but can be returned as `None`. And `job_status` method doesn't do checks.

File: mlcomp/parallelm/extra/sagemaker/monitor/job_monitor_base.py [hunk 4]
```diff
@@ -1,6 +1,8 @@
 import abc
+from datetime import datetime
```
Comment (yakov-g) [inline mlcomp/parallelm/extra/sagemaker/monitor/job_monitor_base.py:2]: Please update CHANGELOG if needed

**Review by yakov-g — APPROVED** [review]

File: mlcomp/parallelm/extra/sagemaker/monitor/job_monitor_base.py (hunk 1 above)
Comment (zohar-mizrahi) [reply mlcomp/parallelm/extra/sagemaker/monitor/job_monitor_base.py:69]: Yes. It is always best to work with UTC and SageMaker times are UTC.

File: mlcomp/parallelm/extra/sagemaker/monitor/report.py (hunk 2 above)
Comment (zohar-mizrahi) [reply mlcomp/parallelm/extra/sagemaker/monitor/report.py:14]: Probably yes. Need to be studied and be defined as a task.

File: mlcomp/parallelm/extra/sagemaker/monitor/job_monitor_base.py (hunk 3 above)
Comment (zohar-mizrahi) [reply mlcomp/parallelm/extra/sagemaker/monitor/job_monitor_base.py:32]: Yes, it is done in`seconds_fmt`, which is called from job_status.

File: mlcomp/parallelm/extra/sagemaker/monitor/job_monitor_base.py (hunk 4 above)
Comment (zohar-mizrahi) [reply mlcomp/parallelm/extra/sagemaker/monitor/job_monitor_base.py:2]: Fixed

**Pull request merged**